tests/swap_over_curve.tests.ts

```diff
@@ -0,0 +1,170 @@
+        partner = Keypair.generate();
+        user = Keypair.generate();
+        poolCreator = Keypair.generate();
+        const receivers = [
+            operator.publicKey,
+            partner.publicKey,
+            user.publicKey,
+            poolCreator.publicKey,
+        ];
+        await fundSol(context.banksClient, admin, receivers);
+        program = createVirtualCurveProgram();
+
+    });
+    it("Swap over the curve", async () => {
+        let totalTokenSupply = 1_000_000_000; // 1 billion
+        let percentageSupplyOnMigration = 10; // 10%;
+        let migrationQuoteThreshold = 300; // 300 sol
+        let tokenBaseDecimal = 6;
+        let tokenQuoteDecimal = 9;
+        let migrationOption = 0; // damm v1
+        let lockedVesting = {
+            amountPerPeriod: new BN(0),
+            cliffDurationFromMigrationTime: new BN(0),
+            frequency: new BN(0),
+            numberOfPeriod: new BN(0),
+            cliffUnlockAmount: new BN(0),
+        };
+        let quoteMint = await createToken(context.banksClient, admin, admin.publicKey, tokenQuoteDecimal);
+        let instructionParams = designCurve(
+            totalTokenSupply,
+            percentageSupplyOnMigration,
+            migrationQuoteThreshold,
+            migrationOption,
+            tokenBaseDecimal,
+            tokenQuoteDecimal,
+            lockedVesting,
+        );
+
+        const params: CreateConfigParams = {
+            payer: partner,
+            leftoverReceiver: partner.publicKey,
+            feeClaimer: partner.publicKey,
+            quoteMint,
+            instructionParams,
+        };
+        let config = await createConfig(context.banksClient, program, params);
+        let configState = await getConfig(context.banksClient, program, config);
+        let swapAmount = instructionParams.migrationQuoteThreshold.mul(new BN(120)).div(new BN(100)); // swap more 20%
+
+        await mintSplTokenTo(context.banksClient, user, quoteMint, admin, user.publicKey, swapAmount.toNumber());
+
+
+        // create pool
+        let virtualPool = await createPoolWithSplToken(context.banksClient, program, {
+            payer: poolCreator,
+            quoteMint,
+            config,
+            instructionParams: {
+                name: "test token spl",
+                symbol: "TEST",
+                uri: "abc.com",
+            },
+        });
+        let virtualPoolState = await getVirtualPool(
+            context.banksClient,
+            program,
+            virtualPool
+        );
+
+        // swap
+        const swapParams: SwapParams = {
+            config,
+            payer: user,
+            pool: virtualPool,
+            inputTokenMint: quoteMint,
+            outputTokenMint: virtualPoolState.baseMint,
+            amountIn: swapAmount,
+            minimumAmountOut: new BN(0),
+            referralTokenAccount: null,
+        };
+        await swap(context.banksClient, program, swapParams);
```

Comment:
can we test how much the user is getting too here?

Reply:
In program, we cap as 20% of migration quote threshold